projects/components/src/filter-button/filter-button.service.ts

```diff
@@ -37,17 +45,29 @@ export class FilterButtonService {
     this.setUrlFilters([...filters]);
   }
 
+  public removeUrlFilter(attributes: FilterAttribute[], filter: Filter): void {
+    const filters = this.getUrlFilters(attributes);
+
+    const foundIndex = filters.findIndex(f => filter.field === f.field);
+
+    if (foundIndex !== -1) {
+      filters.splice(foundIndex, 1);
+    }
+
+    this.setUrlFilters([...filters]);
+  }
+
   private setUrlFilters(filters: Filter[]): void {
     this.navigationService.addQueryParametersToUrl({
       [FilterButtonService.FILTER_QUERY_PARAM]: filters.length === 0 ? undefined : filters.map(f => f.urlString)
     });
   }
 
-  private getUrlFilters(attributes: FilterAttribute[]): Filter[] {
+  public getUrlFilters<T>(attributes: FilterAttribute[]): Filter<T>[] {
```

Comment:
If we want to move the filters out of the URL (which I think we should), then it'd be nice to abstract the add/remove/read methods to remove references to URLs.

Reply:
Agreed. Will do that as part of that effort.